core/collateral/engine.go

```diff
@@ -2176,11 +2181,16 @@ func (e *Engine) getFeeTransferRequest(
 
 		treq.FromAccount = []*types.Account{partyLiquidityFee}
 		treq.ToAccount = []*types.Account{insurancePool}
-		return treq, nil
-
 	default:
 		return nil, ErrInvalidTransferTypeForFeeRequest
 	}
+	// we may be moving funds from the insurance pool, we cannot have more than 1 from account in that case
+	// because once the insurance pool is drained, and a copy of the same account without the updated balance
+	// sits in the FromAccount slice, we are magically doubling the available insurance pool funds.
+	if len(treq.FromAccount) > 0 && treq.FromAccount[0].Type == types.AccountTypeInsurance {
```

Comment:
I don't like the sound of magically doubling funds. Should this not just be a panic if a configuration that shouldn't happen, instead of just fudging and removing the other accounts from the slice?

Feel like there should also be a unittest for this case to make sure we *definitely do not* double anything.